File: basic_algorithms/sort/selectSort/Select.java
package basic_algorithms.sort.selectSort;

import java.util.Random;
import java.util.Arrays;

public class Select {

public static void setSelect(int[] arr) {
    int arrLength = arr.length;

    for (int i=0; i<arrLength; i++) {
        int minIndex = i;

        for (int j=i+1; j<arrLength; j++) {
            if ( arr[j]<arr[minIndex]){
                minIndex = j;
            }
        }
        int tmp = arr[minIndex];
        arr[minIndex] = arr[i];
        arr[i] = tmp;
    } 
}

    public static void main(String[] args) {
        int listLength = 30;
        int[] originalList = new int[listLength];
        Random random = new Random();


        for (int i=0; i<originalList.length; i++) {
            int randNum = random.nextInt(100);
            originalList[i] = randNum;
        }

        setSelect(originalList);
        System.out.println(Arrays.toString(originalList));
    }
    
}

File: sort/selectSort/select.py
import random

def set_selection_sort(arr):
    arr_size = len(arr)

    # 배열을 순회하며 최소값을 선택하여 정렬
    for i in range(arr_size):
        # 현재 위치부터 배열 끝까지 최소값을 찾음
        min_index = i
        for j in range(i+1, arr_size):
            if arr[j] < arr[min_index]:
                min_index = j

        # 현재 위치와 최소값 위치를 교환
        arr[i], arr[min_index] = arr[min_index], arr[i]

if __name__ == "__main__":
    original_list = [random.randint(0,100) for _ in range(100)]

    # 선택 정렬 수행
    set_selection_sort(original_list)

    # 정렬된 결과 출력
    print("정렬된 리스트:", original_list)